test/karma-test-shim.js

```diff
@@ -87,9 +87,38 @@ function configureTestBed() {
 
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
 
-    testing.TestBed.initTestEnvironment(
+    var testBed = testing.TestBed.initTestEnvironment(
       testingBrowser.BrowserDynamicTestingModule,
       testingBrowser.platformBrowserDynamicTesting()
     );
+
+    applyAngularTestingPatches(testBed);
+  });
+}
+
+/** Applies patches and workarounds to Angular's testing package. */
+function applyAngularTestingPatches(testBed) {
+  // Original resetTestingModule function of the TestBed.
+  var _resetTestingModule = testBed.resetTestingModule;
+
```

Comment:
The error that would be thrown is `Cannot read property 'forEach' of undefined`. We have much more knowledge about this failure state, so we can help whoever is debugging this in the future by making a much more helpful message.